tests/regressions.rs

```diff
@@ -79,7 +78,7 @@ fn issue_1205() {
     // spawn a waker thread to wake the poll call below
     let handle = thread::spawn(move || {
         thread::sleep(Duration::from_millis(500));
-        waker1.wake().expect("unable to wake");
+        waker.wake().expect("unable to wake");
```

Comment:
I just realised this change drops the `waker` here and than `Poll` will never get a wakup event (see the `Waker` docs). So this change is invalid.

I'll send a change to ignore the Clippy warning.